cli/tests/CommandExecutorTestRunner.hpp

```diff
@@ -0,0 +1,117 @@
+#include <vector>
+
+#include "parser/SqlParserWrapper.hpp"
+#include "query_execution/Foreman.hpp"
+#include "query_execution/QueryExecutionTypedefs.hpp"
+#include "query_execution/Worker.hpp"
+#include "query_execution/WorkerDirectory.hpp"
+#include "query_execution/WorkerMessage.hpp"
+#include "query_optimizer/tests/TestDatabaseLoader.hpp"
+#include "utility/Macros.hpp"
+#include "utility/textbased_test/TextBasedTestDriver.hpp"
+
+namespace quickstep {
+
+/**
+ * @brief TextBasedTestRunner for testing the ExecutionGenerator.
+ */
+class CommandExecutorTestRunner : public TextBasedTestRunner {
+ public:
+  /**
+   * @brief If this option is enabled, recreate the entire database and
+   * repopulate the data before every test.
+   */
+  static const std::string kResetOption;
+
+  /**
+   * @brief Constructor.
+   */
+  explicit CommandExecutorTestRunner(const std::string &storage_path)
+      : test_database_loader_(storage_path) {
+    test_database_loader_.createTestRelation(false /* allow_vchar */);
+    test_database_loader_.loadTestRelation();
+
+    bus_.Initialize();
+
+    foreman_.reset(new Foreman(&bus_,
+                               test_database_loader_.catalog_database(),
+                               test_database_loader_.storage_manager()));
+    worker_.reset(new Worker(0, &bus_));
+
+    std::vector<client_id> worker_client_ids;
+    worker_client_ids.push_back(worker_->getBusClientID());
+
+    // We don't use the NUMA aware version of foreman code.
+    std::vector<int> numa_nodes;
+    numa_nodes.push_back(-1);
+
+    workers_.reset(new WorkerDirectory(1 /* number of workers */,
+                                       worker_client_ids, numa_nodes));
+    foreman_->setWorkerDirectory(workers_.get());
+
+    worker_->start();
+  }
+
+  ~CommandExecutorTestRunner() {
+    std::unique_ptr<WorkerMessage> poison_message(WorkerMessage::PoisonMessage());
+    TaggedMessage poison_tagged_message(poison_message.get(),
+                                        sizeof(*poison_message),
+                                        quickstep::kPoisonMessage);
+
+    Address worker_address;
+    MessageStyle single_receiver_style;
+
+    worker_address.AddRecipient(worker_->getBusClientID());
+    bus_.Send(foreman_->getBusClientID(),
+              worker_address,
+              single_receiver_style,
+              std::move(poison_tagged_message));
+
+    worker_->join();
+  }
+
+  void runTestCase(const std::string &input,
+                   const std::set<std::string> &options,
+                   std::string *output) override;
+
+ private:
+  SqlParserWrapper sql_parser_;
+  optimizer::TestDatabaseLoader test_database_loader_;
+
+  MessageBusImpl bus_;
+  std::unique_ptr<Foreman> foreman_;
+  std::unique_ptr<Worker> worker_;
+
+  std::unique_ptr<WorkerDirectory> workers_;
+
+  DISALLOW_COPY_AND_ASSIGN(CommandExecutorTestRunner);
+};
+
+
+}  // namespace quickstep
+
+#endif /* QUICKSTEP_CLI_TESTS_COMMAND_EXECUTOR_TEST_RUNNER_HPP_ */
```

Comment:
I know you get this comment from the existing file, but for new file, we should use the following style:

```
#endif  // QUICKSTEP_CLI_TESTS_COMMAND_EXECUTOR_TEST_RUNNER_HPP_
```

Note that two whitespace between `#endif` and `//`.

We should fix the existing files in a separate PR as well.